app/controllers/reservations_controller.rb

```diff
@@ -0,0 +1,52 @@
+class ReservationsController < ApplicationController
+  before_action :set_user
+  before_action :set_reservation, only: %i[show update destroy]
+
+  def index
+    @reservations = @user.reservations
+    render json: @reservations
+  end
+
+  def create
+    @reservation = Reservation.new(reservation_params)
+    @rooms = @reservation.reservation_rooms.find(params[:room_ids])
+
+    if @reservation.save
+      @rooms.each { |room| room.update(reserved: true) }
+      render json: @reservation, status: :created
+    else
+      render json: @reservation.errors, status: :unprocessable_entity
+    end
+  end
```

Comment:
I was having some issues creating and deleting reservations, please check your methods again and make sure they do the actions that are intended. 👍